### Q1. A physician is evaluating a new diet for her patients with a family history of heart disease. To test the effectiveness of this diet, 16 patients are placed on the diet for 6 months. Their weights and triglyceride levels are measured before and after the study, and the physician wants to know if either set of measurements has changed. (Data set: dietstudy.csv)

In [4]:
import numpy as np

import pandas as pd

import matplotlib.pyplot as plt

import scipy.stats as stats

%matplotlib inline

In [7]:
diet = pd.read_csv('G:/Data Science Recorded Lectures/case studies/7. Basic Stats - Case Study-1/7. Basic Stats - Case Study-1/dietstudy.csv')

diet.head()

patid  age  gender  tg0  tg1  tg2  tg3  tg4  wgt0  wgt1  wgt2  wgt3  wgt4
0      1   45    Male  180  148  106  113  100   198   196   193   188   192
1      2   56    Male  139   94  119   75   92   237   233   232   228   225
2      3   50    Male  152  185   86  149  118   233   231   229   228   226
3      4   46  Female  112  145  136  149   82   179   181   177   174   172
4      5   64    Male  156  104  157   79   97   219   217   215   213   214

In [6]:
diet.shape

(16, 13)

### Two Sample T-Test (Paired)

In [4]:
print("The triglyceride levels of patients were {}".format(diet.tg0.mean()))
print("The final triglyceride levels of patients are {}".format(diet.tg4.mean()))
print("The weights of pateints were {}".format(diet.wgt0.mean()))
print("The final weights of pateints are {}".format(diet.wgt4.mean()))

The triglyceride levels of patients were 138.4375
The final triglyceride levels of patients are 124.375
The weights of pateints were 198.375
The final weights of pateints are 190.3125


Stage 1
H0: levels of Triglycerides of individual before diet == levels of Triglycerides of individual after diet


H1: levels of Triglycerides of individual before diet != levels of Triglycerides of individual after diet

In [5]:
stage1_trig = stats.ttest_rel(a=diet.tg0,b=diet.tg4) 

#a =triglyceride levels before diet
#b = triglyceride levels after diet)

In [6]:
stage1_trig.pvalue

0.24874946576903698

In [7]:
stage1_trig.pvalue>0.05

True

Stage 2

H0: weights of individual before diet == weights of individual after diet


H1: weights of individual before diet != weights of individual after diet

In [8]:
stage2_weight = stats.ttest_rel(a=diet.wgt0,b=diet.wgt4) 

#a = weights before diet
#b = weights after diet

In [9]:
stage2_weight.pvalue

1.137689414996614e-08

In [10]:
stage2_weight.pvalue>0.05

False

# Conclusions
Since the p-value for change in weight is less than 0.05,we can conclude that the average loss of 8.06 pounds per patient is not due to chance variation, and can be attributed to the diet.

However, the p-value greater than 0.05 for change in triglyceride level shows the diet did not significantly reduce their triglyceride levels.

### Q2.An analyst at a department store wants to evaluate a recent credit card promotion. To this end, 500 cardholders were randomly selected. Half received an ad promoting a reduced interest rate on purchases made over the next three months, and half received a standard seasonal ad. Is the promotion effective to increase sales?

In [8]:
analyst_credit = pd.read_csv('G:/Data Science Recorded Lectures/case studies/7. Basic Stats - Case Study-1/7. Basic Stats - Case Study-1/creditpromo.csv')

analyst_credit.head()

id         insert      dollars
0   148       Standard  2232.771979
1   572  New Promotion  1403.807542
2   973       Standard  2327.092181
3  1096       Standard  1280.030541
4  1541  New Promotion  1513.563200

In [9]:
standard = analyst_credit['dollars'].loc[analyst_credit['insert']=="Standard"]

promotion = analyst_credit['dollars'].loc[analyst_credit['insert']=="New Promotion"]


In [11]:
print("The average spent by the normal standard seasonal ad is ${}".format(standard.mean()))

print("The average spent by the normal ad is ${}".format(promotion.mean()))

The average spent by the normal standard seasonal ad is $1566.3890309659348
The average spent by the normal ad is $1637.4999830647992


On average, customers who received the interest-rate promotion charged about $70 more than the normal standard season ad, and they vary a little more around their average.

In [12]:
equal_var = stats.ttest_ind(a= standard, b= promotion,equal_var=True)    # equal variance
equal_var.statistic

-2.2604227264649963

In [14]:
unequal_var = stats.ttest_ind(a=standard,b= promotion, equal_var=False)    # Unequal variance
unequal_var.statistic

-2.260422726464996

In [16]:
# Consider equal variance since the t-score is not having a huge difference
unequal_var.statistic - equal_var.statistic

4.440892098500626e-16

In [17]:
t = equal_var.statistic

p = equal_var.pvalue

print(" For the above test, the t-score is {} and the p-value is {}".format(t,p))

if(p<0.05):
    print('We reject null hypothesis')
else:
    print('We fail to reject null hypothesis')

 For the above test, the t-score is -2.2604227264649963 and the p-value is 0.024225996894147814
We reject null hypothesis


Since the significance value of the test is less than 0.05, we can safely conclude that the average of 71.11 dollars more spent by cardholders receiving the reduced interest rate is not due to chance alone. The store will now consider extending the offer to all credit customers.

### Q3. An experiment is conducted to study the hybrid seed production of bottle gourd under open field conditions. The main aim of the investigation is to compare natural pollination and hand pollination. The data are collected on 10 randomly selected plants from each of natural pollination and hand pollination. The data are collected on fruit weight (kg), seed yield/plant (g) and seedling length (cm). (Data set: pollination.csv)
a. Is the overall population of Seed yield/plant (g) equals to 200?

#### One Sample t-Test

In [18]:
pollination = pd.read_csv('G:/Data Science Recorded Lectures/case studies/7. Basic Stats - Case Study-1/7. Basic Stats - Case Study-1/pollination.csv')

pollination.head()

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0  Natural      1.85            147.70            16.86
1  Natural      1.86            136.86            16.77
2  Natural      1.83            149.97            16.35
3  Natural      1.89            172.33            18.26
4  Natural      1.80            144.46            17.90

In [19]:
pollination.Seed_Yield_Plant.mean()

180.8035

In [21]:
ttest_1 = stats.ttest_1samp(a=pollination.Seed_Yield_Plant, popmean = 200)
t_score = round(ttest_1.statistic,2)
p_value = ttest_1.pvalue
print("The p value is {} and the T Score is {}".format(p_value,t_score))

The p value is 0.032891040921283025 and the T Score is -2.3


In [22]:
p_value>0.05

False

Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore overall population of seed yield/plant is not equal to 200.

b. Test whether the natural pollination and hand pollination under open field conditions are equally effective or are significantly different.

#### F-Test/Anova

In [20]:
natural_fruit_wt = pollination['Fruit_Wt'].loc[pollination['Group']=="Natural"]
hand_fruit_wt = pollination['Fruit_Wt'].loc[pollination['Group']=="Hand"]

# Perfrom the Anova
anova = stats.f_oneway(natural_fruit_wt,hand_fruit_wt)
# Statistic :  F Value
f1 = anova.statistic
p1 = anova.pvalue
print("The f-value is {} and the p value is {}".format(f1,p1))


The f-value is 312.228532974426 and the p value is 8.078362076486568e-13


In [22]:
natural_seed_yield = pollination['Seed_Yield_Plant'].loc[pollination['Group']=="Natural"]
hand_seed_yield = pollination['Seed_Yield_Plant'].loc[pollination['Group']=="Hand"]

# Perfrom the Anova
anova = stats.f_oneway(natural_seed_yield,hand_seed_yield)
# Statistic :  F Value
f2 = anova.statistic
p2 = anova.pvalue
print("The f-value is {} and the p value is {}".format(f2,p2))

The f-value is 194.83303662980398 and the p value is 4.271481585484407e-11


In [23]:
natural_seedling_length = pollination['Seedling_length'].loc[pollination['Group']=="Natural"]
hand_seedling_length = pollination['Seedling_length'].loc[pollination['Group']=="Hand"]

# Perfrom the Anova
anova = stats.f_oneway(natural_seedling_length,hand_seedling_length)
# Statistic :  F Value
f3 = anova.statistic
p3 = anova.pvalue
print("The f-value is {} and the p value is {}".format(f3,p3))

The f-value is 6.46293337115627 and the p value is 0.020428817064110556


### 4. An electronics firm is developing a new DVD player in response to customer requests. Using a prototype, the marketing team has collected focus data for different age groups viz. Under 25; 25-34; 35-44; 45-54; 55-64; 65 and above. Do you think that consumers of various ages rated the design differently? (Data set: dvdplayer.csv)

In [25]:
dvd = pd.read_csv('G:/Data Science Recorded Lectures/case studies/7. Basic Stats - Case Study-1/7. Basic Stats - Case Study-1/dvdplayer.csv')

dvd.head()

agegroup   dvdscore
0  65 and over  38.454803
1        55-64  17.669677
2  65 and over  31.704307
3  65 and over  25.924460
4     Under 25  30.450007

### F-Test/Anova

In [26]:
age_group_1 = dvd['dvdscore'].loc[dvd['agegroup']=="65 and over"]
age_group_2 = dvd['dvdscore'].loc[dvd['agegroup']=="55-64"]
age_group_3 = dvd['dvdscore'].loc[dvd['agegroup']=="Under 25"]
age_group_4 = dvd['dvdscore'].loc[dvd['agegroup']=="35-44"]
age_group_5 = dvd['dvdscore'].loc[dvd['agegroup']=="45-54"]
age_group_6 = dvd['dvdscore'].loc[dvd['agegroup']=="25-34"]

# Perfrom the Anova
anova = stats.f_oneway(age_group_1,age_group_2,age_group_3,age_group_4,age_group_5,age_group_6)
# Statistic :  F Value
f = anova.statistic
p = anova.pvalue
print("The f-value is {} and the p value is {}".format(f,p))
if(p<0.05):
    print('We reject null hypothesis')
else:
    print('We fail to reject null hypothesis')

The f-value is 6.992526962676517 and the p value is 3.087324905679639e-05
We reject null hypothesis


Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore,consumers of various ages rated the design differently

### 5. A survey was conducted among 2800 customers on several demographic characteristics. Working status, sex, age, age-group, race, happiness, no. of child, marital status, educational qualifications, income group etc. had been captured for that purpose. (Data set: sample_survey.csv)

a. Is there any relationship in between labour force status with marital status?

## Chi-Square Test

In [27]:
sample_sur = pd.read_csv('G:/Data Science Recorded Lectures/case studies/7. Basic Stats - Case Study-1/7. Basic Stats - Case Study-1/sample_survey.csv')

sample_sur.head()

id            wrkstat        marital  childs   age  educ  paeduc  maeduc  \
0   1  Working full time       Divorced     2.0  60.0  12.0    12.0    12.0   
1   2  Working part-time  Never married     0.0  27.0  17.0    20.0     NaN   
2   3  Working full time        Married     2.0  36.0  12.0    12.0    12.0   
3   4  Working full time  Never married     0.0  21.0  13.0     NaN    12.0   
4   5  Working full time  Never married     0.0  35.0  16.0     NaN    12.0   

   speduc          degree  ...        agecat childcat news1 news2  news3  \
0     NaN     High school  ...      55 to 64      1-2    No    No     No   
1     NaN  Junior college  ...      25 to 34     None    No    No    Yes   
2    16.0     High school  ...      35 to 44      1-2    No    No     No   
3     NaN     High school  ...  Less than 25     None    No    No     No   
4     NaN        Bachelor  ...      35 to 44     None    No    No     No   

  news4 news5      car1      car2      car3  
0    No    No  American  Japanese  Japanese  
1    No    No  American    German  Japanese  
2   Yes   Yes  American  American       NaN  
3   Yes   Yes  American     Other       NaN  
4    No    No  American  American    Korean  

[5 rows x 46 columns]

In [28]:
lab_mar_xtab = pd.crosstab(sample_sur.wrkstat, sample_sur.marital, margins = True)
lab_mar_xtab

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      778            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1345            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1567  
Working part-time         304  
All                      2830

In [29]:
x2test = stats.chi2_contingency(observed= lab_mar_xtab)

x2test

(729.2421426572284,
 1.820339965538765e-127,
 40,
 array([[5.16918728e+01, 1.55886926e+02, 7.68424028e+01, 1.07787986e+01,
         3.28000000e+01, 3.28000000e+02],
        [8.51024735e+00, 2.56643110e+01, 1.26508834e+01, 1.77455830e+00,
         5.40000000e+00, 5.40000000e+01],
        [6.20932862e+01, 1.87254417e+02, 9.23045936e+01, 1.29477032e+01,
         3.94000000e+01, 3.94000000e+02],
        [1.24501767e+01, 3.75459364e+01, 1.85077739e+01, 2.59611307e+00,
         7.90000000e+00, 7.90000000e+01],
        [7.24946996e+00, 2.18621908e+01, 1.07766784e+01, 1.51166078e+00,
         4.60000000e+00, 4.60000000e+01],
        [9.14063604e+00, 2.75653710e+01, 1.35879859e+01, 1.90600707e+00,
         5.80000000e+00, 5.80000000e+01],
        [2.46954770e+02, 7.44740283e+02, 3.67109894e+02, 5.14950530e+01,
         1.56700000e+02, 1.56700000e+03],
        [4.79095406e+01, 1.44480565e+02, 7.12197880e+01, 9.99010601e+00,
         3.04000000e+01, 3.04000000e+02],
        [4.46000000e+02, 1.345

In [30]:
print("The chi square stat is {} and the p value is {}".format(x2test[0],x2test[1]))

The chi square stat is 729.2421426572284 and the p value is 1.820339965538765e-127


#### Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore there is a realtion between labour force status and marital status

#### b. Do you think educational qualification is somehow controlling the marital status?

### Chi- Square Test

In [31]:
edu_mar_xtab = pd.crosstab(sample_sur.degree, sample_sur.marital, margins = True)
edu_mar_xtab

marital         Divorced  Married  Never married  Separated  Widowed   All
degree                                                                    
Bachelor              58      251            129         12       28   478
Graduate              29      123             41          3        9   205
High school          241      686            367         58      148  1500
Junior college        45      108             46          3        6   208
LT High school        70      174             77         17       92   430
All                  443     1342            660         93      283  2821

In [32]:
x2test_b = stats.chi2_contingency(observed= edu_mar_xtab)

x2test_b

(122.68449020508541,
 7.424404099753273e-15,
 25,
 array([[  75.06345268,  227.39312301,  111.83268345,   15.75824176,
           47.95249911,  478.        ],
        [  32.19248493,   97.52215526,   47.9617157 ,    6.75824176,
           20.56540234,  205.        ],
        [ 235.55476781,  713.57674583,  350.9393832 ,   49.45054945,
          150.4785537 , 1500.        ],
        [  32.66359447,   98.94930876,   48.66359447,    6.85714286,
           20.86635945,  208.        ],
        [  67.52570011,  204.55866714,  100.60262318,   14.17582418,
           43.1371854 ,  430.        ],
        [ 443.        , 1342.        ,  660.        ,   93.        ,
          283.        , 2821.        ]]))

In [33]:
print("The chi square stat is {} and the p value is {}".format(x2test_b[0],x2test_b[1]))

The chi square stat is 122.68449020508541 and the p value is 7.424404099753273e-15


Since the significance value of the test is less than 0.05, we reject the null hypothesis. Therefore there educational qualification is somehow controlling the marital status.

c. Is happiness is driven by earnings or marital status?

### Chi-Square Test

### Happiness w.r.t to earnings
H0 : Observed == Expected (No relation between happiness and earnings)

Ha : Observed != Expected (There is a realtionship between happiness and earnings)

In [35]:
happiness_ear_xtab = pd.crosstab(sample_sur.happy, sample_sur.income, margins = True)
happiness_ear_xtab

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   
All                       32             190             178             245   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   
All                      1572             24             32             35   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000   All  
happy                                                                       
Not too happy             14             12              9        11   302  
Pretty happy              13             21             30        13  1408  
Very happy                 6             14             19        11   771  
All                       33             47             58        35  2481

In [36]:
x2test_ear = stats.chi2_contingency(observed= happiness_ear_xtab)

x2test_ear

(178.95053061216427,
 7.234749067043371e-21,
 36,
 array([[   3.89520355,   23.12777106,   21.66706973,   29.82265216,
          191.35187424,    2.92140266,    3.89520355,    4.26037888,
            4.01692866,    5.72108021,    7.06005643,    4.26037888,
          302.        ],
        [  18.16041919,  107.82748892,  101.01733172,  139.04070939,
          892.1305925 ,   13.62031439,   18.16041919,   19.86295848,
           18.72793229,   26.67311568,   32.91575977,   19.86295848,
         1408.        ],
        [   9.94437727,   59.04474002,   55.31559855,   76.13663845,
          488.51753325,    7.45828295,    9.94437727,   10.87666264,
           10.25513906,   14.60580411,   18.0241838 ,   10.87666264,
          771.        ],
        [  32.        ,  190.        ,  178.        ,  245.        ,
         1572.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   58.        ,   35.        ,
         2481.        ]]))

In [37]:
print("The chi square stat of happiness w.r.t to earnings {} and the p value is {}".format(x2test_ear[0],x2test_ear[1]))

The chi square stat of happiness w.r.t to earnings 178.95053061216427 and the p value is 7.234749067043371e-21


#### Happiness w.r.t to marital status
H0 : Observed == Expected (No relation between happiness and marital status)

Ha : Observed != Expected (There is a realtionship between happiness and marital status)

In [38]:
happiness_mar_xtab = pd.crosstab(sample_sur.happy, sample_sur.marital, margins = True)
happiness_mar_xtab

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        278      684            426         49      137  1574
Very happy           93      582            120         13       83   891
All                 443     1337            654         92      279  2805

In [39]:
x2test_mar = stats.chi2_contingency(observed= happiness_mar_xtab)

x2test_mar

(260.68943894182826,
 7.762777322980048e-47,
 15,
 array([[  53.6969697 ,  162.06060606,   79.27272727,   11.15151515,
           33.81818182,  340.        ],
        [ 248.58538324,  750.24527629,  366.98609626,   51.62495544,
          156.55828877, 1574.        ],
        [ 140.71764706,  424.69411765,  207.74117647,   29.22352941,
           88.62352941,  891.        ],
        [ 443.        , 1337.        ,  654.        ,   92.        ,
          279.        , 2805.        ]]))

In [40]:
print("The chi square stat of happiness w.r.t to marital status {} and the p value is {}".format(x2test_mar[0],x2test_mar[1]))

The chi square stat of happiness w.r.t to marital status 260.68943894182826 and the p value is 7.762777322980048e-47


As it can be clearly seen that chi_square score for (happiness w.r.t to marital status) i.e. 260.68 is more than chi_square score for (happiness w.r.t to marital earnings) i.e 178.95, therefore we can conclude that happiness is driven by marital status